src/parser/pegen.ts

```diff
@@ -1502,6 +1521,23 @@ export function seq_flatten(p: Parser, seqs: AST[][]): AST[] {
 //     return _Py_Name(uni, Load, EXTRA_EXPR(first_name, second_name));
 // }
 
+class UnreachableException extends Error {}
+
+export function seq_count_dots(seq: TokenInfo[]) {
+    return seq
+        .map((current_expr) => {
+            switch (current_expr.type) {
+                case ELLIPSIS:
+                    return 3;
+                case DOT:
+                    return 1;
+                default:
+                    throw new UnreachableException();
+            }
+        })
+        .reduce((a, b) => a + b, 0);
```

Comment:
I was again disappointed in javascript for not having a `sum` function

Reply:
would it be the same to do 

```
function _getNumDots(e: TokenInfo) {
    switch (e.type) {
        case ELLIPSIS:
            return 3;
        case DOT:
            return 1;
        default:
            throw new UnreachableException();
    }
}

export function seq_count_dots(seq: TokenInfo[]) {
    return seq.reduce((a, b) => a + _getNumDots(b), 0)
}
```

Reply:
yeah it would, but it is better? 😄

Reply:
it means we don't iterate over the array twice. once for map and once for reduce

Reply:
088b388 - feel free to revert

Reply:
Ah yeah of course I always forget that these are not iterators